File: python_winder/gcode_writer.py
import os


class GCodeWriter:
    def __init__(self, filename: str):
        self.filename = filename

    def write(self, gcode: list[str]) -> None:
        dirpath = os.path.dirname(self.filename)
        if dirpath:
            os.makedirs(dirpath, exist_ok=True)

        with open(self.filename, "w") as f:
            for line in gcode:
                f.write(line + "\n")

File: matlab_winder/GCodeWriter.m
classdef GCodeWriter
    properties
        filename
    end

    methods
        function obj = GCodeWriter(filename)
            obj.filename = filename;
        end

        function write(obj, gcode)
            [pathstr,~,~] = fileparts(obj.filename);

            if ~isempty(pathstr) && ~exist(pathstr, 'dir')
                mkdir(pathstr);
            end

            fid = fopen(obj.filename, 'w');

            if fid == -1
                error("Could not open file: %s", obj.filename);
            end

            for i = 1:length(gcode)
                fprintf(fid, '%s\n', gcode(i));
            end

            fclose(fid);
        end
    end
end
